### DISCUSSION TITLE
User-facing Documentation Ideas

### DISCUSSION THREAD

**Discussion by Coffee2Bits** [DevOps-MBSE/AaC#262] [General]
As we start getting closer to needing to provide user-facing documentation, I figured we could discuss and brainstorm possible content for them.

Topics so far:
* Terminology
  * Definition
  * Model
  * ActiveContext
  * Validator Plugin
* Basic getting started tutorial
  * Produce a minimal model
  * Validate the model
  * Generate some artifact from the model
* Validation (more in-depth on how it works)
* DSL Components (something like when to use the specific root keys)
* How the DSL is impacted by active plugins, extensions, etc
* Extending the DSL
* Using plugins (We can probably hold off on until we have 3rd party plugins or support activating/deactivating plugins)
  * Leveraging AaC APIs in plugins
    * Language Context/Active Context
    * Managing/Manipulating Definitions
* Best Practices
  * Organizing definitions in files
  * Naming conventions (references are camel case and type declarations are CapitializedCamel)

I'll update the list as more come to me.

**Comment by cvchaparro** [comment]
The only thing I'd add is (as we've discussed) some kind of glossary of terms, or something that users can reference to make sure they know what we're talking about. That said, I think we should also make a concerted effort to use language that System Engineers would naturally understand - i.e. use language specific to the System Engineering domain - which can alleviate at least some of this problem but I think we'd still want a defacto location where users can go for a precise definition of what some term means when it's used by us in the project.

**Reply by Coffee2Bits** [reply]
Right! Terminology will be important, but I think we need to engage some system modelers to understand their lingo.

**Comment by Coffee2Bits** [comment]
I think users can benefit from a Best Practices section that includes things like:
* Organizing definitions in files
* Naming conventions (references are camel case and type declarations are CapitializedCamel)

**Comment by jondavid-black** [comment]
I thought it would be good to look at other tool documentation intended for a similar audience.

https://docs.jupyter.org/en/latest/

I like how they have a "Start Here" set of topics right at the front page of their documentation.  The "Get started, "Architecture", "Narratives and Use Cases", and "Installation, Configuration, and Usage" sections would be particularly useful for AaC in my opinion.

I also like how they have "Sub-project Documentation" called out just below the "Get Started" section.  We could do something like that for plug-in documentation.  Perhaps we need to update the gen-plugin plugin to create a /docs folder to contain markdown for plug-in documentation.

**Reply by Coffee2Bits** [reply]
I agree, I like the layout of Jupyter's documentation
